### ISSUE TITLE
CLI Starter Kit Fails on Clean Install with ERR_PNPM_FROZEN_LOCKFILE_WITH_OUTDATED_LOCKFILE

### ISSUE THREAD

**Issue by Hamzehn** [apptension/saas-boilerplate#537]
### Describe the bug

Trying to install for the first time on WSL2 (Windows 11) with the following environment:
- Docker: 26.0.0 (instructions don't specify a min/max required version)
- Node: 18.20.2 (satisfies doc requirement of v18 or above)
- pnpm: 9.0.6 (satisfies doc requirement of v8 or above)
- WSL2 (satisfies doc requirement)
- Python 3.11.9 (satisfies doc requirement of v3.11)

After I fill out the environment variables in the beginning, the installation quickly fails with the following error:
```shell
ERR_PNPM_FROZEN_LOCKFILE_WITH_OUTDATED_LOCKFILE  Cannot perform a frozen installation because the version of the lockfile is incompatible with this version of pnpm

Try either:
1. Aligning the version of pnpm that generated the lockfile with the version that installs from it, or
2. Migrating the lockfile so that it is compatible with the newer version of pnpm, or
3. Using "pnpm install --no-frozen-lockfile".
Installing dependencies... !
    Error: Command failed: pnpm i --frozen-lockfile
```

### Steps to reproduce

1- I made sure I have the following installed on my machine (WSL2 running on Windows)
- Docker: 26.0.0
- Node: 18.20.2
- pnpm: 9.0.6
- WSL2
- Python 3.11.9

2- I created a directory [dir]
```shell
mkdir [dir]
```
3- I ran pnpm create-saas-boilerplate ./[dir]
```shell
pnpm create-saas-boilerplate ./[dir]
```

### System Info

```shell
System:
    OS: Linux 5.15 Ubuntu 20.04.1 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
    Memory: 3.29 GB / 7.64 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 18.20.2 - ~/.nvm/versions/node/v18.20.2/bin/node
    npm: 10.5.0 - ~/.nvm/versions/node/v18.20.2/bin/npm
    pnpm: 9.0.6 - ~/.nvm/versions/node/v18.20.2/bin/pnpm
```

### Logs

<details>
<summary>Click to expand!</summary>

```shell
------ Notice ------
This CLI collects various anonymous events, warnings, and errors to improve the CLI tool and enhance your user experience.
Read more: https://docs.demo.saas.apptoku.com/working-with-sb/dev-tools/telemetry
If you want to opt out of telemetry, you can set the environment variable SB_TELEMETRY_DISABLED to 1 in your shell.
For example:
   export SB_TELEMETRY_DISABLED=1

Performing system check:
✔ Checking if docker is installed... it is!
✔ Checking if pnpm is installed... it is!
Project will be initialized in directory: /home/[***]/[***]
Fetching latest release... done
Latest release: 2.6.0
Downloading project... done
Setup environment variables:
This step will set up all environment variables.
For more details visit: https://docs.demo.saas.apptoku.com/api-reference/env.

Pass PROJECT_NAME [saas]: saas
Configure you social OAuth clients (optional):
Pass SOCIAL_AUTH_GOOGLE_OAUTH2_KEY [<CHANGE_ME>]:
Pass SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET [<CHANGE_ME>]:
Pass SOCIAL_AUTH_FACEBOOK_KEY [<CHANGE_ME>]:
Pass SOCIAL_AUTH_FACEBOOK_SECRET [<CHANGE_ME>]:
Configure Stripe access (optional):
Pass STRIPE_LIVE_SECRET_KEY [<CHANGE_ME>]:
Pass STRIPE_TEST_SECRET_KEY [<CHANGE_ME>]:
Pass DJSTRIPE_WEBHOOK_SECRET [<CHANGE_ME>]:
Pass STRIPE_PUBLISHABLE_KEY [<CHANGE_ME>]: 
Configure Contentful access (optional):
Pass CONTENTFUL_SPACE_ID [<CHANGE_ME>]:
Pass CONTENTFUL_ACCESS_TOKEN [<CHANGE_ME>]:
Pass CONTENTFUL_ENVIRONMENT [<CHANGE_ME>]:
Installing dependencies...
packages/webapp                          |  WARN  The field "pnpm" was found in /home/[***]/[***]/packages/webapp/package.json. This will not take effect. You should configure "pnpm" at the root of the workspace instead.
Scope: all 22 workspace projects
 ERR_PNPM_FROZEN_LOCKFILE_WITH_OUTDATED_LOCKFILE  Cannot perform a frozen installation because the version of the lockfile is incompatible with this version of pnpm

Try either:
1. Aligning the version of pnpm that generated the lockfile with the version that installs from it, or
2. Migrating the lockfile so that it is compatible with the newer version of pnpm, or
3. Using "pnpm install --no-frozen-lockfile".
Installing dependencies... !
    Error: Command failed: pnpm i --frozen-lockfile
```
</details>

### Validations

- [X] Follow our [Code of Conduct](https://github.com/apptension/saas-boilerplate/blob/master/CODE_OF_CONDUCT.md).
- [X] Read the [Contributing Guidelines](https://github.com/apptension/saas-boilerplate/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://docs.demo.saas.apptoku.com/).
- [X] Check that there isn't [already an issue](https://github.com/apptension/saas-boilerplate/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/apptension/saas-boilerplate/discussions) or join our [Discord Chat Server](https://discord.apptension.com).

**Comment by mkleszcz** [comment]
Thanks for creating this issue! It looks like there might be a problem with backward compatibility of the pnpm 9+. Will check this

**Comment by kun391** [comment]
I had the same issue when I used the pnpm version 7.20.0

```
Warning: run-commands command "docker-compose build workers" exited with non-zero status code
 
 >  NX   Running target compose-build-image for 2 projects and 2 tasks they depend on failed
   Failed tasks:
   
   - workers:compose-build-image
Warning: run-commands command "nx run-many --target=compose-build-image --projects=backend,workers" exited with non-zero status code


 >  NX   Ran target docker-compose:up for project core and 2 task(s) they depend on (5s)
 
    ✖    1/3 failed
    ✔    2/3 succeeded [1 read from cache]
 
    Error: "pnpm nx run core:docker-compose:up" failed with code 1
```

Resolved with pnpm version 8.15.7
